portal/src/components/analytics/DeleteSubscriptionDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation, useQueryClient } from "react-query";
+import { SubscriptionsService } from "../../services";
+import useMoonToast from "../../hooks/useMoonToast";
+import useAnalytics from "../../contexts/AnalyticsContext";
+import styles from "./DeleteSubscriptionDialog.module.css";
+import { Flex, Modal, ModalContent, Text, Spinner } from "@chakra-ui/react";
+
+const DeleteSubscriptionDialog = ({
+  isOpen,
+  onClose,
+  contract,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  contract: { label: string; id: string };
+}) => {
+  const toast = useMoonToast();
+  const { setSelectedAddressId } = useAnalytics();
+  const queryClient = useQueryClient();
+  const deleteSubscription = useMutation(SubscriptionsService.deleteSubscription(), {
+    onError: (error: Error) => toast(error.message, "error"),
+    onSuccess: () => {
+      setSelectedAddressId(0);
+      queryClient.invalidateQueries("subscriptions"); //TODO direct update
```

Comment:
What is the status of this TODO? If we want merge without it, I think we should create an issue and remove the TODO.

Reply:
Why? I like TODOs. And we use them a lot - I have seen them in every repo I look at - api, web3, spire etc. Actually we have hundreds in our codebase. In the portal we have 41 now. At least most of them don't qualify as github issues